samza-api/src/main/java/org/apache/samza/operators/triggers/Trigger.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.operators.triggers;
+
+
+import org.apache.samza.operators.data.MessageEnvelope;
+
+/**
+ * Marker interface for all triggers. The firing of a trigger indicates the completion of a window pane.
+ *
+ * <p> Use the {@link Triggers} APIs to create a {@link Trigger}.
+ *
+ * @param <M> the type of the incoming {@link MessageEnvelope}
+ * @param <K> the type of key in {@link MessageEnvelope}
+ * @param <V> the type of value in the {@link MessageEnvelope}
+ */
+public interface Trigger<M extends MessageEnvelope, K, V> {
```

Comment:
Gave it some more thought. K and V are not meaningful as parameters for Trigger classes. The only case where M is meaningful is the specification of a token based trigger. (which we'll add later). All triggers now take M as the parameter.